cmd/check_tx_lookup.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+   Copyright 2021 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include <CLI/CLI.hpp>
+
+#include <boost/filesystem.hpp>
+#include <silkworm/etl/collector.hpp>
+#include <silkworm/common/log.hpp>
+#include <silkworm/db/access_layer.hpp>
+#include <silkworm/db/util.hpp>
+#include <silkworm/crypto/ecdsa.hpp>
+#include <silkworm/db/tables.hpp>
+#include <silkworm/db/stages.hpp>
+#include <silkworm/types/transaction.hpp>
+#include <boost/endian/conversion.hpp>
+#include <silkworm/common/util.hpp>
+#include <silkworm/chain/config.hpp>
+#include <iostream>
+
+using namespace silkworm;
+ 
+int main(int argc, char* argv[]) { 
+    namespace fs = boost::filesystem;
+    
+    CLI::App app{"Generates Blockhashes => BlockNumber mapping in database"};
+
+    std::string db_path{db::default_path()};
+    app.add_option("-d,--datadir", db_path, "Path to a database populated by Turbo-Geth", true)
+        ->check(CLI::ExistingDirectory);
+    CLI11_PARSE(app, argc, argv);
+
+    Logger::default_logger().set_local_timezone(true);  // for compatibility with TG logging
+
+    // Check data.mdb exists in provided directory
+    boost::filesystem::path db_file{boost::filesystem::path(db_path) / boost::filesystem::path("data.mdb")};
+    if (!boost::filesystem::exists(db_file)) {
+        SILKWORM_LOG(LogError) << "Can't find a valid TG data file in " << db_path << std::endl;
+        return -1;
+    }
+    fs::path datadir(db_path);
+    fs::path etl_path(datadir.parent_path() / fs::path("etl-temp"));
+    fs::create_directories(etl_path);
+    etl::Collector collector(etl_path.string().c_str(), /* flush size */ 512 * kMebi);
+
+    lmdb::DatabaseConfig db_config{db_path};
+    db_config.set_readonly(false); 
+    std::shared_ptr<lmdb::Environment> env{lmdb::get_env(db_config)};
+    std::unique_ptr<lmdb::Transaction> txn{env->begin_rw_transaction()};
```

Comment:
As you're only checking no need to set db in rw mode and also you can simply open a ro transaction